drivers/char/hw_random/exynos_trng_rust.rs

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! RNG driver for Exynos TRNGs
+//!
+//! Author: Maciej Falkowski <m.falkowski@samsung.com>
+//!
+//! Copyright 2021 (c) Samsung Electronics Software, Inc.
+//!
+//! Based on the Exynos TRNG driver drivers/char/hw_random/exynos-rng by
+//! Łukasz Stelmach <l.stelmach@samsung.com>
+
+#![no_std]
+#![feature(allocator_api, global_asm)]
+#![allow(dead_code)]
```

Comment:
If one needs to allow a lint, we should have a comment explaining why.

Reply:
This lint was placed to suppress the warnings of [unused global constants L37-L57](https://github.com/Rust-for-Linux/linux/pull/554/files#diff-2893e507cf97fa126e30036f8ace781308a3140071c093a98aaf13de1dc532aaR37-R57).

Leaving them with underscore should be okay?
```diff
-const EXYNOS_TRNG_ONLINE_CTRL: usize = 0x40;
-const EXYNOS_TRNG_ONLINE_STAT: usize = 0x44;
-const EXYNOS_TRNG_ONLINE_MAXCHI2: usize = 0x48;
+const _EXYNOS_TRNG_ONLINE_CTRL: usize = 0x40;
+const _EXYNOS_TRNG_ONLINE_STAT: usize = 0x44;
+const _EXYNOS_TRNG_ONLINE_MAXCHI2: usize = 0x48;
 const EXYNOS_TRNG_FIFO_CTRL: usize = 0x50;
 const EXYNOS_TRNG_FIFO_0: usize = 0x80;
-const EXYNOS_TRNG_FIFO_1: usize = 0x84;
-const EXYNOS_TRNG_FIFO_2: usize = 0x88;
-const EXYNOS_TRNG_FIFO_3: usize = 0x8c;
-const EXYNOS_TRNG_FIFO_4: usize = 0x90;
-const EXYNOS_TRNG_FIFO_5: usize = 0x94;
-const EXYNOS_TRNG_FIFO_6: usize = 0x98;
-const EXYNOS_TRNG_FIFO_7: usize = 0x9c;
+const _EXYNOS_TRNG_FIFO_1: usize = 0x84;
+const _EXYNOS_TRNG_FIFO_2: usize = 0x88;
+const _EXYNOS_TRNG_FIFO_3: usize = 0x8c;
+const _EXYNOS_TRNG_FIFO_4: usize = 0x90;
+const _EXYNOS_TRNG_FIFO_5: usize = 0x94;
+const _EXYNOS_TRNG_FIFO_6: usize = 0x98;
+const _EXYNOS_TRNG_FIFO_7: usize = 0x9c;
```

Reply:
Yeah, self-documentation purpose only. Also it would be nice to provide 1:1 C and Rust versions, for code comparison.

Reply:
In that case, I would use the underscore -- it is clear and local, which allows `dead_code` to still work for everything else. 

(Sometimes another option is grouping the unused ones in a module etc., but it would not look right here).

Perhaps you can add a comment somewhere about why they are there, in any case, e.g.:

```rust
// Unused constants are kept for documentation purposes
// and to keep the driver as close to the C version as possible.
```

Reply:
Regarding the constants in general, I would consider using namespaces (e.g. modules or enums) if it makes sense.

By the way, I think we don't gain much by repeating `EXYNOS_TRNG` either.

Reply:
> In that case, I would use the underscore -- it is clear and local, which allows `dead_code` to still work for everything else.
> 
> (Sometimes another option is grouping the unused ones in a module etc., but it would not look right here).
> 
> Perhaps you can add a comment somewhere about why they are there, in any case, e.g.:
> 
> ```rust
> // Unused constants are kept for documentation purposes
> // and to keep the driver as close to the C version as possible.
> ```
Sure, I will apply it

> Regarding the constants in general, I would consider using namespaces (e.g. modules or enums) if it makes sense.
> 
> By the way, I think we don't gain much by repeating `EXYNOS_TRNG` either.

Yeah, both solutions would work, with enums I propose:
```rust
enum TrngReg {
    OnlineCtrl = 0x40,
    OnlineStat = 0x44,
    OnlineMmaxChi2 = 0x48,
}
```
using modules:
```rust
mod trng_reg {
    pub const ONLINE_CTRL: usize = 0x40;
    pub const ONLINE_STAT: usize = 0x44;
    pub const ONLINE_MAXCHI2: usize = 0x48;
}
```
For simple number constants, I think that enum is not worth the effort.
I'd vote for keeping the names as they are so the code is more interoperable(easier porting and grepping) with C's side.

I'm eager to hear your response.